website/docs/guides/customize-schema-alias.md

```diff
@@ -0,0 +1,490 @@
+
+:::
+
+
+### 2. Static schemas - Adding developer identities to the front of tables instead of the schema
+
+Occasionally, we run into instances where the security posture of the organization prevents developers from creating schemas and all developers have to develop in a single schema.
+
+In this case, we can: 
+
+- Change `generate_schema_name()` to use a single schema for all developers, even if a custom schema is set.
+- Update `generate_alias_name()` to append the developer alias and the custom schema to the front of the table name in the dev environment.
+    - This method is not ideal, as it can cause long table names, but it will let developers see in which schema the model will be created in production.
+
+```Jina
+
+{% macro generate_alias_name(custom_alias_name=none, node=none) -%}
+
+    {%- if  env_var('DBT_ENV_TYPE','DEV') == 'DEV' -%}
+
+        {%- if custom_alias_name -%}
+
+            {{ target.schema }}__{{ custom_alias_name | trim }}
+
+        {%- elif node.version -%}
+
+            {{ target.schema }}__{{ node.name ~ "_v" ~ (node.version | replace(".", "_")) }}
+
+        {%- else -%}
+
+            {{ target.schema }}__{{ node.name }}
+
+        {%- endif -%}
+    
+    {%- else -%}
+
+        {%- if custom_alias_name -%}
+
+            {{ custom_alias_name | trim }}
+
+        {%- elif node.version -%}
+
+            {{ return(node.name ~ "_v" ~ (node.version | replace(".", "_"))) }}
+
+        {%- else -%}
+
+            {{ node.name }}
+
+        {%- endif -%}
+
+    {%- endif -%}
+
+{%- endmacro %}
+
+```
+
+```Jina
+
+{% macro generate_schema_name(custom_schema_name, node) -%}
+
+    {%- set default_schema = target.schema -%}
+    {%- if custom_schema_name is none -%}
+
+        {{ default_schema }}
+
+    {%- elif  env_var('DBT_ENV_TYPE','DEV') != 'CI' -%}
+        
+        {{ custom_schema_name | trim }}
+
+    {%- else -%}
+
+        {{ default_schema }}_{{ custom_schema_name | trim }}
+
+    {%- endif -%}
+
+{%- endmacro %}
+
+```
+
+This will generate the following outputs for a model called `my_model` with a custom schema of `marketing`, preventing any overlap of objects between dbt runs from different contexts.
+
+
+| Context     |Target database| Target schema | Resulting object               |
+|-------------|:-------------:|:-------------:|:------------------------------:|
+| Developer 1 | dev           | dbt_dev1      |dev.marketing.dbt_dev1_my_model |
+| Developer 2 | dev           | dbt_dev2      |dev.marketing.dbt_dev2_my_model  |
+| CI PR 123   | ci            | dbt_pr_123    |ci.dbt_pr_123_marketing.my_model|
+| CI PR 234   | ci            | dbt_pr_234    |ci.dbt_pr_234_marketing.my_model|
+| Production  | prod          | analytics     |prod.marketing.my_model         |
+
+
+### 3. Using the branch name for the schema prefix, instead of the developer identity
```

Comment:
```suggestion
### 3. Use branch name for the schema prefix
```

Reply:
should we reiterate the instructions and tell the user to add these in their macros folder. Is it a separate file for each macro? i think so yea?

e.g.:
Create a new file called generate_schema_name.sql under your macros folder with the following code: